### PR TITLE
[AE-75] Add tutorial for IoT Cloud with MicroPython

### PR DESCRIPTION
**Pull request by sebromero**
## What This PR Changes
- This PR adds a new tutorial that explains how to use the MicroPython library to access Arduino IoT Cloud

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    client.start()
+
+```
+
+**Explanations:**
+
+- `wifi_connect()` Connects to your local WiFi using the credentials specified in secrets.py.
+- `client.register` Registers a variable that will be synced with the cloud.
+- `on_switch_changed` Is the callback that gets executed when the `ledSwitch` variable is changed by toggling the switch on the cloud dashboard. This function in turn toggles the on-board LED and updates the cloud variable `led` that reflects the state of the on-board LED to be displayed in the cloud dashboard.
+- `client.start()`Enters a loop that runs as long as the board is connected to the cloud and synchronises data as it runs.
```
Comment (karlsoderby) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md:169]:
```suggestion
- `wifi_connect()` - Connects to your local WiFi using the credentials specified in secrets.py.
- `client.register` - Registers a variable that will be synced with the cloud.
- `on_switch_changed` - Is the callback that gets executed when the `ledSwitch` variable is changed by toggling the switch on the cloud dashboard. This function in turn toggles the on-board LED and updates the cloud variable `led` that reflects the state of the on-board LED to be displayed in the cloud dashboard.
- `client.start()` - Enters a loop that runs as long as the board is connected to the cloud and synchronises data as it runs.
```

**Review by karlsoderby — CHANGES_REQUESTED** [review]
Only minor changes, rest looks great.

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+## Introduction 
+
+This tutorial guides you on how to use the MicroPython library to connect your Arduino device to the Arduino IoT Cloud.
+
+## Goals
+
+The goals of this tutorial are:
+
+- Connect your Arduino device to your Wi-Fi network.
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md:21]:
```suggestion
- Connect your Arduino device to your Wi-Fi® network.
```

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+- [Arduino Lab for MicroPython](https://labs.arduino.cc/en/labs/micropython)
+- [Arduino GIGA R1 WiFi](/hardware/giga-r1-wifi) or [Portenta H7](/hardware/portenta-h7)
+- MicroPython >= 1.2 installed on your Arduino device (download firmware [here](/micropython/#firmware)).
+
+
+
+## Setting Up Your Device and IoT Cloud
+
+Before we start, make sure you have MicroPython installed on your board. If you haven't you can follow [this tutorial](https://docs.arduino.cc/micropython/basics/board-installation). Then configure a "thing" in the [Arduino IoT Cloud](https://create.arduino.cc/iot/) consisting of two boolean variables called `led` and `ledSwitch`. To set up a "thing" and a corresponding dashboard, please follow these two tutorials:
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md:35]:
```suggestion
Before we start, make sure you have MicroPython installed on your board. If you haven't you can follow [this tutorial](https://docs.arduino.cc/micropython/basics/board-installation). Then configure a Thing in the [Arduino IoT Cloud](https://create.arduino.cc/iot/) consisting of two boolean variables called `led` and `ledSwitch`. To set up a Thing and a corresponding dashboard, please follow these two tutorials:
```

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+- [IoT Cloud Variables](https://docs.arduino.cc/arduino-cloud/getting-started/cloud-variables)
+- [IoT Cloud Dashboards & Widgets](https://docs.arduino.cc/arduino-cloud/getting-started/dashboard-widgets)
+
+The resulting thing and dashboard should look similar to the following:
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md:40]:
```suggestion
The resulting Thing and dashboard should look similar to the following:
```

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+![Thing with two boolean variables.](./assets/thing.png)
+
+![Dashboard with an LED and a Switch widget.](./assets/dashboard.png)
+
+Also, your device needs to be registered. Follow the flow "Any Device" ("Manual") when clicking "Add" in the "Devices" tab.
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md:46]: Please use bold when you are highlighting button names (if that is what these in " " are)

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+Give your board the desired name.
+
+![Give your device a name of your choice.](./assets/set-device-name.png)
+
+Eventually write down the Device ID / Secret Key pair that you will need to connect your device to Arduino IoT Cloud.
+
+![The connection credentials to be written down.](./assets/get-key.png)
+
+You will obtain a pair of device id and device key after registration. Store these details, along with your Wi-Fi credentials, in a `secrets.py` file. Here is an example of how `secrets.py` should look like:
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md:58]:
```suggestion
You will obtain a pair of device id and device key after registration. Store these details, along with your Wi-Fi® credentials, in a `secrets.py` file. Here is an example of how `secrets.py` should look like:
```

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+    client.register("led", value=None)
+
+    # Start the Arduino IoT cloud client.
+    client.start()
+
+```
+
+**Explanations:**
+
+- `wifi_connect()` - Connects to your local WiFi using the credentials specified in secrets.py.
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md:166]:
```suggestion
- `wifi_connect()` - Connects to your local Wi-Fi® using the credentials specified in secrets.py.
```

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+Open Arduino Lab for MicroPython and connect to your board. Pasting the above code and run the script. Then open your Arduino IoT Cloud dashboard. You should see the registered "ledSwitch" and "led" widgets. Toggle the "ledSwitch", and the LED on your Arduino board should light up accordingly. The state of the "led" variable should also change, mirroring the state of the physical LED.
+
+
+
+## Troubleshoot
+
+If the code is not working, there are some common issues we can troubleshoot:
+
+- Make sure MicroPython >= 1.2 is installed on your board.
+- Check the Wi-Fi credentials in the `secrets.py` file.
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md:184]:
```suggestion
- Check the Wi-Fi® credentials in the `secrets.py` file.
```

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+- Ensure the device ID and Cloud password in the `secrets.py` file match with what is registered on the IoT Cloud.
+- Make sure your IoT Cloud "thing" is correctly set up and your device is assigned to it.
```
Comment (jhansson-ard) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md:186]:
```suggestion
- Make sure your IoT Cloud Thing is correctly set up and your device is assigned to it.
```

**Review by jhansson-ard — CHANGES_REQUESTED** [review]
I left a couple of quick spell- and trademark checks ⭐

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md (hunk 5 above)
Comment (sebromero) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md:46]:
```suggestion
Also, your device needs to be registered. Follow the flow "Any Device" ("Manual") when clicking **Add** in the "Devices" tab.
```

**Review by karlsoderby — APPROVED** [review]
LGTM!

File: content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md (hunk 5 above)
Comment (sebromero) [inline content/arduino-cloud/01.getting-started/10.iot-cloud-micropython/content.md]:
```suggestion
Also, your device needs to be registered. Follow the flow "Any Device" ("Manual") when clicking **Add** in the "Devices" tab.
```

**Pull request merged**